doc/warm-reboot/view_switch.md

```diff
@@ -0,0 +1,179 @@
+# View Switching in ProducerStateTable
+
+### Problem Description
+To support warm reboot, 
+ProducerStateTable needs to support a feature that producer can create a temporary view for a table, 
+into which write operation will not not be synced to consumers immediately.
+Instead, the producer can explicitly ask to apply the view, 
+upon which all the objects in new view will get synced to consumer.
+
+However, the actual operations get passed to consumer should be optimized as best effort, 
+so if an object is present in both old view and new view, 
+it should not be `del()` and `set()` again.
+
+In order to archieve that, we are going to add two additional APIs: `create_temp_view()` and `apply_temp_view()`,
+as well as modifying existing `del()` and `set()` API.
+
+This design is based on the assumption that there is only one producer writing to a specific table. 
+If there are multiple producer writing to a same table, the changes that 'secondary' producer write to the table
+while 'main' producer was doing view switching might be lost.
+
+
+### Redis Objects and Memory Objects
+As a reference, here are the exiting Redis objects corresponding to a certain table:
+
+| Terminology   | Sample Redis Object  |
+| ------------- | :-----------------:  |
+| TableHash     | ` ROUTE_TABLE:25.78.106.0/27` |
+| StateHash     | `_ROUTE_TABLE:25.78.106.0/27` |
+| KeySet        | `ROUTE_TABLE_KEY_SET`      |
+| DelKeySet     | `ROUTE_TABLE_DEL_SET`      |
+| Channel       | `ROUTE_TABLE_CHANNEL`      |
+
+We are not going to create additional Redis object to support the temporary view. 
+Instead, an in-memory table `TableDump m_tempViewState` 
+that shares a similar structure with TableHash and StateHash will be created to maintain the target state of the temporary view.
+Additionally, a flag `bool m_tempViewActive` will indicates whether a temporary view is current being worked on or not.
+
+### Operation Pseudo Code
```

Comment:
Can we add a working flow of the API caller?  Producer calling flow and Consumer calling flow?